content/docs/glossary/sponsored-reserves.mdx

```diff
@@ -85,8 +58,248 @@ See [Revoke Sponsorship](../start/list-of-operations.mdx#revoke-sponsorship) for
+)
+
+func main() {
+	client := sdk.DefaultTestNetClient
+
+	// Both S1 and S2 will be sponsors for A at various points in time.
+	S1, A, S2 := keypair.MustRandom(), keypair.MustRandom(), keypair.MustRandom()
+	sponsoredAccount := A.Address()
+
+	for _, pair := range []*keypair.Full{S1, A, S2} {
+		resp, err := http.Get("https://friendbot.stellar.org/?addr=" + pair.Address())
+		if err != nil {
+			panic(err)
+		}
+		resp.Body.Close()
+		fmt.Println("Funded", pair.Address())
+	}
+
+	//
+	// NOTE: Error checks are omitted for brevity; always check return values!
+	//
+
+	// Load the corresponding account for both A and C.
+	s1Account, err := client.AccountDetail(sdk.AccountRequest{AccountID: S1.Address()})
+	aAccount, err := client.AccountDetail(sdk.AccountRequest{AccountID: A.Address()})
+	s2Account, err := client.AccountDetail(sdk.AccountRequest{AccountID: S2.Address()})
+
+	// Arbitrary assets to sponsor trustlines for. Let's assume they make sense.
+	assets := []txnbuild.CreditAsset{
+		txnbuild.CreditAsset{Code: "ABCD", Issuer: S1.Address()},
+		txnbuild.CreditAsset{Code: "EFGH", Issuer: S1.Address()},
+		txnbuild.CreditAsset{Code: "IJKL", Issuer: S2.Address()},
+	}
+}
+```
+
+</CodeExample>
+
+
+### Sponsoring a Trustline
+Now, let's sponsor trustlines for Account A. Notice how the `CHANGE_TRUST` operation is sandwiched between the begin and end sponsoring operations and that all relevant accounts need to sign the transaction.
+
+<CodeExample>
+
+```go
+	//
+	// 1. S1 will sponsor a trustline for Account A.
+	//
+	sponsorTrustline := []txnbuild.Operation{
+		&txnbuild.BeginSponsoringFutureReserves{
+			SourceAccount: &s1Account,
+			SponsoredID:   sponsoredAccount,
+		},
+		&txnbuild.ChangeTrust{
+			SourceAccount: &aAccount,
+			Line:          &assets[0],
+			Limit:         txnbuild.MaxTrustlineLimit,
+		},
+		&txnbuild.EndSponsoringFutureReserves{
+			SourceAccount: &aAccount,
+		},
+	}
+
+	// Note that while A can submit this transaction, both sign it.
+	SignAndSend(client, &aAccount, []*keypair.Full{S1, A}, sponsorTrustline...)
+	fmt.Println("Sponsored a trustline of", A.Address())
 
-https://github.com/stellar/stellar-protocol/blob/master/core/cap-0033.md#example-revoke-sponsorship
+	//
+	// 2. Both S1 and S2 sponsor trustlines for Account A for different assets.
+	//
+	sponsorTrustline = []txnbuild.Operation{
+		&txnbuild.BeginSponsoringFutureReserves{
+			SourceAccount: &s1Account,
+			SponsoredID:   sponsoredAccount,
+		},
+		&txnbuild.ChangeTrust{
+			SourceAccount: &aAccount,
+			Line:          &assets[1],
+			Limit:         txnbuild.MaxTrustlineLimit,
+		},
+		&txnbuild.EndSponsoringFutureReserves{},
+
+		&txnbuild.BeginSponsoringFutureReserves{
+			SourceAccount: &s2Account,
+			SponsoredID:   sponsoredAccount,
+		},
+		&txnbuild.ChangeTrust{
+			SourceAccount: &aAccount,
+			Line:          &assets[2],
+			Limit:         txnbuild.MaxTrustlineLimit,
+		},
+		&txnbuild.EndSponsoringFutureReserves{},
```

Comment:
It looks like the transaction source account is `aAccount`, so the source account here doesn't have to be specified. I noticed in some of your other examples you do specify the source account even if it's equal to the transaction source account. Might be a good idea to keep the usage consistent.

Reply:
Good call, I'll consolidate this.

Reply:
Addressed, I believe